packages/ragbits-core/src/ragbits/core/vector_stores/in_memory.py

```diff
@@ -25,7 +25,7 @@ def __init__(
             metadata_store: The metadata store to use.
         """
         super().__init__(default_options=default_options, metadata_store=metadata_store)
-        self._storage: dict[str, VectorStoreEntry] = {}
+        self._storage: list[VectorStoreEntry] = []
```

Comment:
I'm not sure about this change, with the list we allow duplicates in storage

Reply:
That's intentional. I see preventing multiple documents with the same textual key as a bug. Textual representations are not unique. We can have multiple different images with the same description (or even: no description, as described in #175). And even for textual documents - the store shouldn't silently loose documents just because they have the same content as other document. They have different metadata and point to different sources. If we wanted to add them all to the store, there was a reason for that, and it's not for the store to decide.

Reply:
I agree, which is why we have open ticket #160 to address this. Your fix doesn't solve the problem we have with Chroma vector store, where we also can't load duplicate keys due to collisions. 

There is also a risk that a store implemented as a list may grow indefinitely with the same ingest calls (very unlikely, but not impossible). My take is to leave it as it is and solve #160 and we should be good.

Reply:
Then #160 would become a blocker for merging this PR because the script would become non-functional - only the single last image would ever be stored.

Reply:
(and it's also unclear to me how generating a random uuid would help with the same objects being ingested multiple time - as far as I understand, each digestion would simply get a new random uuid)

Reply:
yeah, uuid doesn't solve the problem here, but changing the components of sha256 should help (add element location data)